aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/AWSCognitoAuthPlugin.java

```diff
@@ -510,6 +515,99 @@ public void onError(Exception exception) {
         }
     }
 
+    @Override
+    public void rememberDevice(
+            @NonNull Action onSuccess,
+            @NonNull Consumer<AuthException> onException
+    ) {
+        awsMobileClient.getDeviceOperations().updateStatus(true, new Callback<Void>() {
+            @Override
+            public void onResult(Void result) {
+                onSuccess.call();
+            }
+
+            @Override
+            public void onError(Exception exception) {
+                onException.accept(new AuthException(
+                        "An error occurred while remembering a device",
+                        exception,
+                        "See attached exception for more details"
+                ));
+            }
+        });
+    }
+
+    @Override
+    public void forgetDevice(
+            @NonNull Action onSuccess,
+            @NonNull Consumer<AuthException> onException
+    ) {
+        awsMobileClient.getDeviceOperations().forget(new Callback<Void>() {
+            @Override
+            public void onResult(Void result) {
+                onSuccess.call();
+            }
+
+            @Override
+            public void onError(Exception exception) {
+                onException.accept(new AuthException(
+                        "An error occurred while forgetting a device",
+                        exception,
+                        "See attached exception for more details"
+                ));
+            }
+        });
+    }
+
+    @Override
+    public void forgetDevice(
+            @NonNull AuthDevice device,
+            @NonNull Action onSuccess,
+            @NonNull Consumer<AuthException> onException
+    ) {
+        awsMobileClient.getDeviceOperations().forget(device.getDeviceId(), new Callback<Void>() {
+            @Override
+            public void onResult(Void result) {
+                onSuccess.call();
+            }
+
+            @Override
+            public void onError(Exception exception) {
+                onException.accept(new AuthException(
+                        "An error occurred while forgetting a device",
+                        exception,
+                        "See attached exception for more details"
+                ));
+            }
+        });
+    }
+
+    @Override
+    public void fetchDevices(
+            @NonNull Consumer<List<AuthDevice>> onSuccess,
+            @NonNull Consumer<AuthException> onException
+    ) {
+        awsMobileClient.getDeviceOperations().list(new Callback<ListDevicesResult>() {
+            @Override
+            public void onResult(ListDevicesResult result) {
+                List<AuthDevice> devices = new ArrayList<>();
+                for (Device device : result.getDevices()) {
+                    devices.add(AuthDevice.fromId(device.getDeviceKey()));
```

Comment:
Did you verify we can't get the device names from AWSMobileClient?

Reply:
Will verify with jithin or someone else before I push this!

Reply:
https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/Auth/AWSCognitoAuthPlugin/Support/Utils/Device%2BExtension.swift
Swift appears to be ignoring name when creating a device also.